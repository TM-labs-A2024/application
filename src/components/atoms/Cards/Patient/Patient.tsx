import { WarningIcon } from '@chakra-ui/icons'
import { Link } from '@chakra-ui/next-js'
import { Text, Tag, TagLeftIcon, TagLabel, Tooltip } from '@chakra-ui/react'
import NextLink from 'next/link'
import React from 'react'

import Chevron from '../../../../../public/static/icons/chevron.svg'

export default function Patient({
  href,
  title,
  description,
  status,
  hospitalizationPlace
}: {
  href: string
  title: string
  description: string
  status?: string
  hospitalizationPlace?: string
}) {
  return (
    <Link as={NextLink} href={href} className="w-full" data-testid="patient-card">
      <div className="flex flex-row items-center justify-between">
        <div className="flex flex-col items-start gap-1">
          <Text fontSize="md" as="b">
            {title}
          </Text>
          <Text>{description}</Text>
        </div>
        {status === 'hospitalizado' && (
          <Tooltip label={hospitalizationPlace}>
            <Tag size="md" key={status} variant="subtle" colorScheme="gray">
              <TagLeftIcon boxSize="12px" as={WarningIcon} />
              <TagLabel>{status}</TagLabel>
            </Tag>
          </Tooltip>
        )}
        <Chevron />
      </div>
    </Link>
  )
}
