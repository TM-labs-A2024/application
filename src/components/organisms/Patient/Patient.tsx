'use client'
import { Text, Heading, Stack, Link } from '@chakra-ui/react'
import { format } from 'date-fns'
import { es } from 'date-fns/locale/es'
import NextLink from 'next/link'
import React from 'react'

export default function Patient({
  patient
}: {
  patient: {
    id: number
    name: string
    birthdate: string
    email: string
    phone: string
  }
}) {
  // --- Hooks -----------------------------------------------------------------
  // --- END: Hooks ------------------------------------------------------------

  // --- Local state -----------------------------------------------------------
  // --- END: Local state ------------------------------------------------------

  // --- Refs ------------------------------------------------------------------
  // --- END: Refs -------------------------------------------------------------

  // --- Redux -----------------------------------------------------------------
  // --- END: Redux ------------------------------------------------------------

  // --- Side effects ----------------------------------------------------------
  // --- END: Side effects -----------------------------------------------------

  // --- Data and handlers -----------------------------------------------------
  // --- END: Data and handlers ------------------------------------------------

  return (
    <div className="mx-auto block h-screen w-screen overflow-hidden px-8 pt-8 lg:w-1/2">
      <div className="mb-8 flex w-full flex-row justify-between">
        <Heading as="h2" size="md" noOfLines={1}>
          Perfil
        </Heading>
        <Link as={NextLink} href="/perfil/editar">
          Editar
        </Link>
      </div>
      <Stack spacing={1} mb={6}>
        <Heading as="h3" size="md" noOfLines={1}>
          {patient.name}
        </Heading>
        <Text>CI: {patient.id.toLocaleString('es-ES')}</Text>
      </Stack>
      <Stack mb={6}>
        <h4 className="text-sm text-gray-600">Fecha de nacimiento</h4>
        <Text className="font-medium">
          {format(new Date(patient.birthdate), "dd 'de' MMMM, yyyy", {
            locale: es
          })}
        </Text>
      </Stack>
      <Stack mb={6}>
        <h4 className="text-sm text-gray-600">Correo electrónico</h4>
        <Text className="font-medium">{patient.email}</Text>
      </Stack>
      <Stack mb={6}>
        <h4 className="text-sm text-gray-600">Teléfono</h4>
        <Text className="font-medium">{patient.phone}</Text>
      </Stack>
    </div>
  )
}
