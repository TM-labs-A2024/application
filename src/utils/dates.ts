export const formatDate = (date: string) => {
  const chunks = date.split('-')

  const newDate = chunks[2] + '-' + chunks[1] + '-' + chunks[0]

  return newDate
}
