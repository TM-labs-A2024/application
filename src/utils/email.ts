import emailjs from '@emailjs/browser'

export function sendEmail(data: {
  from_name: string
  to_name: string
  code: string | number
  to_email: string
}) {
  const templateParams = {
    ...data
  }

  const serviceKey = process.env.NEXT_PUBLIC_EMAIL_SERVICE_KEY ?? ''
  const templateKey = process.env.NEXT_PUBLIC_EMAIL_TEMPLATE_KEY ?? ''
  const apiKey = process.env.NEXT_PUBLIC_EMAIL_API_KEY ?? ''

  emailjs.send(serviceKey, templateKey, templateParams, apiKey).catch(() => {
    alert('Algo sali√≥ mal, por favor reintenta el registro.')
  })
}
