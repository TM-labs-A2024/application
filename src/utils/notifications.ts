import { GENERIC_ERROR } from '@constants/index'
import { isMobile } from '@utils/index'
import { Store } from 'react-notifications-component'

export const setupErrorNotification = () => {
  Store.addNotification(
    GENERIC_ERROR('Algo salió mal, por favor inténtelo de nuevo.', isMobile(window))
  )
}
