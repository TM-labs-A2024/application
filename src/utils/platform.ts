import { Capacitor } from '@capacitor/core'

export const isWeb = () => Capacitor.getPlatform() === 'web'

export const isIOS = () => Capacitor.getPlatform() === 'ios'

export const isAndroid = () => Capacitor.getPlatform() === 'android'
