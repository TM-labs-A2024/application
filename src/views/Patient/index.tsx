import Patient from '@components/organisms/Patient'
import Layout from '@layouts/index'
import React from 'react'

export default function PatientView({
  patient
}: {
  patient: {
    id: number
    firstname: string
    lastname: string
    birthdate: string
    email: string
    phone: string
  }
}) {
  return (
    <Layout>
      <Patient patient={patient} />
    </Layout>
  )
}
