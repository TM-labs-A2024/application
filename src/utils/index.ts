import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
