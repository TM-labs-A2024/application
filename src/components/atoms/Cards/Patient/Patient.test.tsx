import { render, screen } from '@test/utils/index'

import '@testing-library/jest-dom'
import Patient from './Patient'

const patient = {
  href: '/evolucion/3fa85f64-5717-4562-b3fc-2c963f66afa5/1/1',
  title: 'Evolución 1: 17, septiembre 2024',
  description: 'Editado por: Pedro Pérez',
  status: 'Hospitalizado'
}

const notHospizatlizedPatient = {
  href: '/evolucion/3fa85f64-5717-4562-b3fc-2c963f66afa5/1/1',
  title: 'Evolución 1: 17, septiembre 2024',
  description: 'Editado por: Pedro Pérez',
  status: ''
}

describe('Atoms > Cards > Patient test', () => {
  test('The card shows the title, description and status of the patient', async () => {
    render(<Patient {...patient} />)

    expect(await screen.findByText(patient.title)).toBeInTheDocument()
    expect(await screen.findByText(patient.description)).toBeInTheDocument()
    expect(await screen.findByText(patient.status)).toBeInTheDocument()
  })

  test("The card doesn't show the status of the patient if not hospitalized", async () => {
    render(<Patient {...notHospizatlizedPatient} />)

    expect(await screen.findByText(notHospizatlizedPatient.title)).toBeInTheDocument()
    expect(await screen.findByText(notHospizatlizedPatient.description)).toBeInTheDocument()
    expect(screen.queryByText(patient.status)).not.toBeInTheDocument()
  })

  test('The url matches the expected', async () => {
    render(<Patient {...patient} />)

    const enlace = await screen.findByRole('link')
    expect(enlace).toHaveAttribute('href', patient.href)
  })

  test('Matches the snapshot', async () => {
    render(<Patient {...patient} />)

    expect(await screen.findByTestId('patient-card')).toMatchSnapshot()
  })
})
