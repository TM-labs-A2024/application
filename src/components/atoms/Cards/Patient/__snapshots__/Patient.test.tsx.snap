// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms > Cards > Patient test Matches the snapshot 1`] = `
<a
  class="chakra-link w-full css-spn4bz"
  data-testid="patient-card"
  href="/evolucion/3fa85f64-5717-4562-b3fc-2c963f66afa5/1/1"
>
  <div
    class="flex flex-row items-center justify-between"
  >
    <div
      class="flex flex-col items-start gap-1"
    >
      <b
        class="chakra-text css-qqfgvy"
      >
        Evolución 1: 17, septiembre 2024
      </b>
      <p
        class="chakra-text css-0"
      >
        Editado por: Pedro Pérez
      </p>
    </div>
    <span
      class="css-1eveppl"
    >
      <svg
        class="chakra-icon chakra-icon css-vvjt0k"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
          fill="currentColor"
        />
      </svg>
      <span
        class="css-1ny2kle"
      >
        hospitalizado
      </span>
    </span>
    <div />
  </div>
</a>
`
